virttest/virt_vm.py

```diff
@@ -668,6 +670,19 @@ def get_address(self, index=0):
                 be for each wait_for_login() call.
                 """
                 devs = set([nic.netdst])
+            if not arp_ip:
+                if (len(macs) > 1) and (flexible_index == "yes"):
+                    # Try to poke for each index
+                    for i in range(len(macs)):
+                        try:
+                            mac = self.get_virsh_mac_address(i)
+                        except IndexError:
+                            break
```

Comment:
Checked utils_env.py and 'utils_net' it looks we save MAC address in lowercase, I think "self.address_cache.get(mac.lower())" is enough.